File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>>map = new HashMap<>();
        for(String str:strs){
            char[] sortedChar = str.toCharArray();
            Arrays.sort(sortedChar);
            if(!map.containsKey(String.valueOf(sortedChar))){
                map.put(String.valueOf(sortedChar), new ArrayList<>());
            }
            System.out.println(String.valueOf(sortedChar));
            map.get(String.valueOf(sortedChar)).add(str);
        }
      List<List<String>>ans = new ArrayList<>();
      for(String st: map.keySet()){
        ans.add(map.get(st));
      }
      return ans;
    }
}

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>>ans;
        map<string,vector<string>>map;
        int count[26];
        for(string s:strs){
            string key = s;
            sort(key.begin(), key.end());
            map[key].push_back(s);
        }
        //auto& p:map
        for(auto p:map){
            ans.push_back(std::move(p.second));
        }
        return ans;
    }
};